packages/devops/scripts/ci/validateBranchName.sh

```diff
@@ -1,11 +1,32 @@
 #!/bin/bash
 
-reserved_endings=(api config export mobile admin www)
+DIR=`dirname "$0"`
+. ${DIR}/utils.sh
 
-for reserved_ending in ${reserved_endings[@]}
+MAX_LENGTH=56
+RESERVED_ENDINGS=(api config export mobile admin www)
+
+branch_name="$CI_BRANCH"
+if [[ $branch_name == "" ]]; then
+    # Get currently checked out branch
+    branch_name=`git rev-parse --abbrev-ref HEAD`
+fi
+
+# Validate branch name length
+name_length=${#branch_name}
+if [[ $name_length -gt MAX_LENGTH ]]; then
+    log_error "❌ Branch name is too long, must be $MAX_LENGTH characters max"
+    exit 1;
+fi
+
+# Validate branch name ending
+for reserved_ending in ${RESERVED_ENDINGS[@]}
 do
-    if [[ "$CI_BRANCH" == *$reserved_ending ]]; then
-        echo "Invalid branch name, must not end in '$reserved_ending'"
+    if [[ "$branch_name" == *$reserved_ending ]]; then
+        log_error "❌ Invalid branch name ending: '$reserved_ending'"
         exit 1;
     fi
 done
+
+log_success "✔ Branch name is valid!"
+exit 0;
```

Comment:
I think there are a couple more restrictions on branch names:
- No uppercase letters
- Characters which may interfere with URL namespaces must not be included (eg. `/ \ . & ?` etc.)
- A couple more namespace reserved endings: `aggregation`, `psss`?)

https://docs.beyondessential.com.au/books/software-development/page/git-branching-strategy-and-naming-conventions

Reply:
Oh great call! I will update the functionality accordingly

Reply:
Do you have an idea of how I could find a comprehensive list of

> Characters which may interfere with URL namespaces

If not, I believe using `/ \ . & ?` would be good enough for now - we can always update the list if need be

Reply:
I think I got that list just from googling which characters would need to be escaped for a URL originally hehe :P

I think that's fine for now, if anyone encounters others they can just update the script. It's not common for anyone to include those special chars in their branch names anyway

Reply:
@kael89 is it possible to check if the branch has upper case letters? If I recall upper case letters causes failure re-deploying to feature branches

Reply:
Sorry, my bad - it should definitely be possible! I just missed that 👍

Reply:
@rohan-bes ready to re-review